### ISSUE TITLE
Problem with "agent-js-cypress" integration

### ISSUE THREAD

**Issue by notfounnd** [reportportal/agent-js-cypress#93]
Hi.

I'm having problems integrating Cypress and ReportPortal (5.5.0 and 5.3.5).

When using the "autoMerge" flag to merge the runs, the ReportPortal does not record (or appears not to be receiving the logs) of test runs.

Without using the option it logs one execution per file, as expected.

A validation I also performed was to check the behavior of the example project available here on GitHub. The problem occurred the same way (verified with Cypress versions 7.2, 7.7, 8.0 and 8.6).

I tried to change the strategy and follow the option "isLaunchMergeRequired" and I wasn't successful either (the guidelines presented in the repository caused an error in Cypress).

Another detail is that the assertions made by Cypress are not being registered in the scenario summary in the ReportPortal (the entries in the RP only show the hooks in a disorganized way). The example below demonstrates this behavior.

```javascript
/// <reference types="cypress" />

describe('Example context test', { tags: ['@debug', '@example'] }, () => {
  before(() => {
    cy.log('Before suite test')
  })

  beforeEach(() => {
    cy.log('Before each test')
  })

  afterEach(() => {
    cy.log('After each test')
  })

  after(() => {
    cy.log('After suite test')
  })

  it('Should use assertions', () => {
    const target = '_blank'

    cy.visit('/commands/assertions')
    cy.exampleAssertTable()
    cy.exampleAssertLinks(target)
  })

  it('Should use a fixture', () => {
    cy.exampleServiceIntercept().as('getComment')

    cy.visit('/commands/files')
    cy.get('.fixture-btn').click()
    cy.wait('@getComment').its('response.body')
      .should('have.property', 'name')
      .and('include', 'Using fixtures to represent data')
  })
})
```

![image](https://user-images.githubusercontent.com/57534340/138112360-72e7b45e-8fa5-48b6-8f20-b8c1c7e2fcd9.png)

**Comment by chivekrodis** [comment]
Hi @notfoundd, thanks for the request. At the moment the agent does not seem to work correctly with the cypress-grep lib. If possible, try running the tests without using cypress-grep and see the result. You can also use the example folder to run a test with logs inside the hooks. Everything seems to be reporting correctly. Please write if you managed to run the tests without errors

**Comment by notfounnd** [comment]
Hi @chivekrodis, thanks for your feedback. I’ll check out that tip.

I have an observation about this info: the demo repository does not have cypress-grep installed and it presented the same behavior.

- https://github.com/reportportal/examples-js/blob/master/example-cypress/package.json

**Comment by notfounnd** [comment]
Hi. I created a new project using only the following dependencies:

- cypress
- agent-js-cypress 

It presented same preveously behavior. Validating with another example repo (e.g. Jest), it works as expected.

![image](https://user-images.githubusercontent.com/57534340/154694673-a597e77b-5bc7-4e9f-868e-f01a3b30e888.png)

![image](https://user-images.githubusercontent.com/57534340/154694906-94cdacff-0798-4814-a867-fb729d3f9723.png)

![image](https://user-images.githubusercontent.com/57534340/154695111-e3e36b3f-4ee1-425e-b07f-1216af574e2c.png)

![image](https://user-images.githubusercontent.com/57534340/154696103-91691978-e80f-4069-8aab-f57370f75c01.png)

**Comment by notfounnd** [comment]
@brabenetz in the plugin/index.js file I set the second parameter.

I didn't see this "trick" about timeout. I will do and I return about that try here.

![image](https://user-images.githubusercontent.com/57534340/154725197-acb36f20-6c4e-439e-a3c6-b0dd52c1a407.png)

**Comment by notfounnd** [comment]
@brabenetz thanks for tip.

After I set the option 'restClientConfig' in the cypress.json file the integration worked.

```
"restClientConfig": {
   "timeout": 300000
}
```

**Comment by AM1988** [comment]
@notfounnd could you please tell us if you have an issue with cypress": "9.5.3" and "@reportportal/agent-js-cypress": "5.0.3" that when the test execution has been finished the test run in RP is still in progress? (like in the attached screenshot)  
![image](https://user-images.githubusercontent.com/19313323/162730902-4b2a699c-c023-42e9-9c42-30ae9439fae9.png)

@brabenetz I've applied all your tips, thank you.

**Comment by notfounnd** [comment]
@AM1988 I've had an similiar issue like it inside on our CI/CD process execution. In my troubleshooting I took a chance about the end of communication between CI/CD and RP.

Pipeline was killing virtual machine instance before it send all messages from test execution to RP.

I solved this issue adding an a explicit wait (30 seconds) on my pipeline script execution.

```yaml
...
   script:
     - cd /app
     - yarn run:example
     - sleep 30
...
```
